includes/User.php

```diff
@@ -671,7 +671,7 @@ public static function isCreatableName( $name ) {
 
 		// Preg yells if you try to give it an empty string
 		if( $wgInvalidUsernameCharacters !== '' ) {
-			if( preg_match( '/[' . preg_quote( $wgInvalidUsernameCharacters, '/' ) . ']/', $name ) ) {
+			if( preg_match( $wgInvalidUsernameCharacters, $name ) ) {
```

Comment:
Let's do it as you proposed, this will allow us to get rid of the additional code more easily if needed.